src/cmap/connection.ts

```diff
@@ -124,7 +124,15 @@ export interface GetMoreOptions extends CommandOptions {
   batchSize?: number;
   maxTimeMS?: number;
   maxAwaitTimeMS?: number;
-  comment?: Document | string;
+  /**
+   * Comment to apply to the operation.
+   *
+   * In server versions pre-4.4, 'comment' must be string.  A server
+   * error will be thrown if any other type is provided.
+   *
+   * In server versions 4.4 and above, 'comment' can be any valid BSON type.
+   */
+  comment?: any;
```

Comment:
I'll tackle the second question first.  `undefined` is not permitted (well, it'll be permitted by typescript but not included as a comment) and null is allowed.  `null` is the preferred nullish value in bson and `undefined` is deprecated as a bson type.  We have to draw the line somewhere though, since `comment` is an optional property we have to have some way to let users not specify a comment.  I think permitting any value but `undefined` is reasonable.  I could add a note in the docstring though to make it explicit.  

The second question is more interesting and Neal might know better than I do.  I considered creating a `BSONLike` type that encompassed all possible BSON values that we support.  But in my experimentation, I couldn't find a type or construct that BSON wouldn't serialze.

Reply:
per slack conversation, let's update to `unknown`

Reply:
Done!